### PR TITLE
[SYCL][UR] Add UR bindings 

### PR DESCRIPTION
**Pull request by jandres742**

### CODE REVIEW

**Comment by jandres742** [comment]
To merge after https://github.com/intel/llvm/pull/8315

**Comment by bader** [comment]
@jandres742, I restarted pre-commit checks. OpenCL CPU results were invalid due to infrastructure issue. Sorry for inconvenience.

NOTE: failure on GPU seems to be related to the change though.

**Comment by jandres742** [comment]
Rebased.

**Comment by jandres742** [comment]
@smaslov-intel please review.

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_common.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+//===--------- ur_level_zero.hpp - Level Zero Adapter -----------------===//
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_common.cpp:1]: I will update these headers for all files with the each file's name.

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_program.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+#pragma once
+
+#include <cassert>
+#include <list>
+#include <map>
+#include <stdarg.h>
```
Comment (igchor) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_program.hpp:13]: nit: do we need all those includes (e.g. stdarg.h)? Probably some of the could be moved to .cpp file or even removed?

**Review by igchor — COMMENTED** [review]
LGTM

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_program.hpp (hunk 2 above)
Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_program.hpp:13]: thanks @igchor . Not actually needed at the moment,  I can remove them and then add them as needed.

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_native.cpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//===--------- ur_level_zero_native.cpp - Level Zero Adapter ----------===//
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_native.cpp:1]: What will "native" be for?

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 4]
```diff
@@ -101,7 +101,29 @@ add_sycl_plugin(level_zero
     "../unified_runtime/ur/usm_allocator_config.cpp"
     "../unified_runtime/ur/usm_allocator_config.hpp"
     "../unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp"
+    "../unified_runtime/ur/adapters/level_zero/ur_level_zero_common.hpp"
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:104]: looks you are missing "kernel" here, is that intentional?

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_native.cpp (hunk 3 above)
Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_native.cpp:1]: not sure at the moment. in the local development I have been doing I haven't used this file actually. I can remove it and add it later if needed.

File: sycl/plugins/level_zero/CMakeLists.txt (hunk 4 above)
Comment (jandres742) [reply sycl/plugins/level_zero/CMakeLists.txt:104]: right, kernel code would go in the module file, at least for now.

**Review by smaslov-intel — APPROVED** [review]

**Comment by jandres742** [comment]
> LGTM

@igchor : addressed comments. Please check.

**Comment by smaslov-intel** [comment]
@intel/llvm-reviewers-runtime : please review/approve/merge this

**Pull request merged**